Modules/_ctypes/cfield.c

```diff
@@ -232,25 +287,101 @@ PyCField_get(PyObject *op, PyObject *inst, PyTypeObject *type)
     }
     src = _CDataObject_CAST(inst);
     return PyCData_get(st, self->proto, self->getfunc, inst,
-                       self->index, self->size, src->b_ptr + self->offset);
+                       self->index, _pack_legacy_size(self),
+                       src->b_ptr + self->byte_offset);
 }
 
 static PyObject *
-PyCField_get_offset(PyObject *self, void *data)
+PyCField_get_legacy_size(PyObject *self, void *data)
 {
-    return PyLong_FromSsize_t(_CFieldObject_CAST(self)->offset);
+    CFieldObject *field = _CFieldObject_CAST(self);
+    return PyLong_FromSsize_t(_pack_legacy_size(field));
 }
 
 static PyObject *
-PyCField_get_size(PyObject *self, void *data)
+PyCField_get_bit_size(PyObject *self, void *data)
 {
-    return PyLong_FromSsize_t(_CFieldObject_CAST(self)->size);
+    CFieldObject *field = _CFieldObject_CAST(self);
+    if (field->bitfield_size) {
+        return PyLong_FromSsize_t(field->bitfield_size);
+    }
+    if (field->byte_size < PY_SSIZE_T_MAX / 8) {
+        return PyLong_FromSsize_t(field->byte_size * 8);
+    }
+    // If the bit size overflows Py_ssize_t, we don't try fitting it in
+    // a bigger C type. Use Python ints.
+    PyObject *byte_size_obj = PyLong_FromSsize_t(field->byte_size);
+    if (!byte_size_obj) {
+        return NULL;
+    }
+    PyObject *eight = PyLong_FromLong(8);
+    if (!eight) {
+        return NULL;
+    }
+    return PyNumber_Multiply(byte_size_obj, eight);
+
+}
+
+static PyObject *
+PyCField_is_bitfield(PyObject *self, void *data)
+{
+    return PyBool_FromLong(_CFieldObject_CAST(self)->bitfield_size);
+}
+
+static PyObject *
+PyCField_is_anonymous(PyObject *self, void *data)
+{
+    return PyBool_FromLong(_CFieldObject_CAST(self)->anonymous);
 }
 
 static PyGetSetDef PyCField_getset[] = {
-    { "offset", PyCField_get_offset, NULL, PyDoc_STR("offset in bytes of this field") },
-    { "size", PyCField_get_size, NULL, PyDoc_STR("size in bytes of this field") },
-    { NULL, NULL, NULL, NULL },
+    { "size", PyCField_get_legacy_size, NULL,
+        PyDoc_STR("size in bytes of this field. For bitfields, this is a "
+                  "legacy packed value; use byte_size instead") },
+
+    { "bit_size", PyCField_get_bit_size, NULL,
+        PyDoc_STR("size of this field in bits") },
+    { "is_bitfield", PyCField_is_bitfield, NULL,
+        PyDoc_STR("true if this is a bitfield") },
+    { "is_anonymous", PyCField_is_anonymous, NULL,
+        PyDoc_STR("true if this field is anonymous") },
+    { NULL },
+};
+
+static PyMemberDef PyCField_members[] = {
+    { "name",
+        .type=Py_T_OBJECT_EX,
```

Comment:
Nit but I think we write `.member = ...` with surrounding spaces around `=` for clarity.